tsfc/driver.py

```diff
@@ -328,7 +328,10 @@ def compile_expression_dual_evaluation(expression, to_element, coordinates, inte
     if domain:
         assert coordinates.ufl_domain() == domain
         builder.domain_coordinate[domain] = coordinates
-        builder.set_cell_sizes(domain)
+        if domain.ufl_cell().topological_dimension() != 0:
+            # Only P0 is a valid finite element if topological_dimension
+            # is 0 so set_cell_sizes is unnecessary
+            builder.set_cell_sizes(domain)
```

Comment:
Why not do the appropriate thing in the builder? Keep the logic close to the implementation.

Reply:
done in 9a46507296f28504666bfee290c3068ebe1c9ec0